lib/public/views/Runs/format/formatRunDetectors.js

```diff
@@ -29,23 +29,51 @@ export const formatRunDetectors = (run, detailsModel) => {
         return '-';
     }
 
+    const _detectorTypes = detailsModel.detectorTypes.match({
+        NotAsked: () => [],
+        Loading: () => [],
+        Success: (data) => data,
+        Failure: (error) => error,
+    });
+
+    const _allAndSelectedDetectorTypes = _detectorTypes.map(({ id, name }) => {
+        const isPresent = detectorsQualities.map((obj) => obj.id).includes(id);
+        const quality = isPresent ? detectorsQualities.filter((obj) => obj.id == id)[0].quality : null;
+        return { id, name, quality, isPresent };
+    });
+
     const detectorsBadges = h(
         '.g2.flex-row.flex-grow.flex-wrap.justify-end',
-        detectorsQualities.map(({ id, name, quality }) => {
+        _allAndSelectedDetectorTypes.map(({ id, name, quality, isPresent }) => {
```

Comment:
This code works fine, but I would suggest another implementation. The objective is to loop only once over the list of all detectors, and once over the list of detectors actually linked to the run:

Instead of the actual `_allAndSelectedDetectorTypes`, I would create a map indexing the detector quality (value) by the detector id (name).
Then, here, I would loop over all the detectors, and get the quality from the map (if it's undefined, it means that the detector is not included)